compiler/rustc_mir/src/transform/instrument_coverage.rs

```diff
@@ -638,6 +656,10 @@ impl<'a, 'tcx> Instrumentor<'a, 'tcx> {
     ///    into a broader combined span (because it represents a contiguous, non-branching, and
     ///    uninterrupted region of source code).
     ///
+    ///    Closures are exposed in their enclosing functions as `Assign` `Rvalue`s, and since
```

Comment:
I wonder if this will always be true? Is there a more generic way we can query for a span being part of another function?

Ironically, I think this would be easier in HIR

Reply:
I don't think this can change, but if for some reason it did, it would be a significant change in Rust and the MIR, and whoever changes it would have to fix this as well. (Tests would catch it.) 

I did consider several alternatives, but I think this is the cleanest approach.

One alternative (that I rejected) would be to query the module for all other functions, get their spans, and see if any of those spans are inside the span of the function being instrumented. That is much more complex (both conceptually and from a performance perspective).

I'm quite happy with the current implementation actually.